lsp/ada_ls.lua

```diff
@@ -0,0 +1,29 @@
+local util = require 'lspconfig.util'
+
+---@brief
+---
+---https://github.com/AdaCore/ada_language_server
+--
+-- Installation instructions can be found [here](https://github.com/AdaCore/ada_language_server#Install).
+--
+-- Workspace-specific [settings](https://github.com/AdaCore/ada_language_server/blob/master/doc/settings.md) such as `projectFile` can be provided in a `.als.json` file at the root of the workspace.
+-- Alternatively, configuration may be passed as a "settings" object to `vim.lsp.config('ada_ls', {})`:
+--
+-- ```lua
+-- vim.lsp.config('ada_ls', {
+--     settings = {
+--       ada = {
+--         projectFile = "project.gpr";
+--         scenarioVariables = { ... };
+--       }
+--     }
+-- })
+-- ```
+return {
+  cmd = { 'ada_language_server' },
+  filetypes = { 'ada' },
+  root_dir = function(bufnr, done_callback)
```

Comment:
```suggestion
  root_dir = function(bufnr, on_done)
```

Reply:
Unfortunately I've used `done_callback` in every configs that needed it. Should I rename every occurrence?

Reply:
yes that's always implied :)

 I went ahead and pushed this change.